src/shims/time.rs

```diff
@@ -85,6 +87,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         Ok(0)
     }
 
+    #[allow(non_snake_case)]
+    fn GetSystemTimeAsFileTime(&mut self, LPFILETIME_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os("windows", "GetSystemTimeAsFileTime");
+        this.check_no_isolation("GetSystemTimeAsFileTime")?;
+
+        let NANOS_PER_SEC = this.eval_windows_u64("time", "NANOS_PER_SEC")?;
+        let INTERVALS_PER_SEC = this.eval_windows_u64("time", "INTERVALS_PER_SEC")?;
+        let INTERVALS_TO_UNIX_EPOCH = this.eval_windows_u64("time", "INTERVALS_TO_UNIX_EPOCH")?;
+        let NANOS_PER_INTERVAL = NANOS_PER_SEC / INTERVALS_PER_SEC;
+        let SECONDS_TO_UNIX_EPOCH = INTERVALS_TO_UNIX_EPOCH / INTERVALS_PER_SEC;
+
+        let duration = system_time_to_duration(&SystemTime::now())?
```

Comment:
This will fail if `rustc` is running on a Windows system with its clock configured to a time before the unix epoch.

Reply:
That is already true however for e.g. Unix targets running on Windows hosts. The error in that case will be "times before the Unix epoch are not supported", which is reasonably accurate. And the usecase is sufficiently niche that I am okay with not supporting this initially.

(Patches/bugreports still welcome, of course.)

Reply:
For unix targets on Windows hosts, I would personally prefer saturation as a solution instead.

Reply:
Don't many targets support negative times though? Linux at least does, AFAIK. So some times before the epoch could still be supported, we just don't, for now -- because why would we spend any effort on systems with their clock before 1970?^^

Reply:
Also I'd prefer to fail loudly over silently doing the wrong thing (like saturation).